src/store/modules/gov.ts

```diff
@@ -101,11 +101,20 @@ const actions = {
         balances = balances.map(balance =>
           parseFloat(formatUnits(balance.toString(), decimals))
         );
+        Object.fromEntries(
+            Object.entries(proposals).map(async (proposal: any, i) => {
+                const proposalObj = await dispatch('getProposal', {
```

Comment:
This will be complicated to scale, now the page load all the proposals, if there is 500 proposals there will be thousand of requests.

Reply:
yes, thats why i said earlier we better backend it